builtInFunctions/migrateDataTrie_test.go

```diff
@@ -89,32 +92,150 @@ func TestMigrateDataTrie_ProcessBuiltinFunction(t *testing.T) {
 			},
 		}
 
-		mdtf, _ := NewMigrateDataTrieFunc(vmcommon.BuiltInCost{}, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{})
+		mdtf, _ := NewMigrateDataTrieFunc(vmcommon.BuiltInCost{}, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{}, &mock.ShardCoordinatorStub{})
 		vmOutput, err := mdtf.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), mock.NewUserAccount([]byte("dest")), input)
 		assert.Nil(t, vmOutput)
 		assert.Equal(t, ErrBuiltInFunctionCalledWithValue, err)
 	})
-
-	t.Run("nil dest account", func(t *testing.T) {
+	t.Run("invalid number of arguments", func(t *testing.T) {
 		t.Parallel()
 
 		input := &vmcommon.ContractCallInput{
 			VMInput: vmcommon.VMInput{
 				CallValue: big.NewInt(0),
+				Arguments: [][]byte{[]byte("arg1"), []byte("arg2")},
+			},
+		}
+
+		mdtf, _ := NewMigrateDataTrieFunc(vmcommon.BuiltInCost{}, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{}, &mock.ShardCoordinatorStub{})
+		vmOutput, err := mdtf.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), nil, input)
+		assert.Nil(t, vmOutput)
+		assert.True(t, errors.Is(err, ErrInvalidNumberOfArguments))
+
+		input.Arguments = [][]byte{}
+		vmOutput, err = mdtf.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), nil, input)
+		assert.Nil(t, vmOutput)
+		assert.True(t, errors.Is(err, ErrInvalidNumberOfArguments))
+	})
+	t.Run("tx is not to self", func(t *testing.T) {
+		t.Parallel()
+
+		input := &vmcommon.ContractCallInput{
+			VMInput: vmcommon.VMInput{
+				CallValue:  big.NewInt(0),
+				Arguments:  [][]byte{[]byte("arg1")},
+				CallerAddr: []byte("caller"),
+			},
+			RecipientAddr: []byte("recipient"),
+		}
+
+		mdtf, _ := NewMigrateDataTrieFunc(vmcommon.BuiltInCost{}, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{}, &mock.ShardCoordinatorStub{})
+		vmOutput, err := mdtf.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), nil, input)
+		assert.Nil(t, vmOutput)
+		assert.Equal(t, ErrOperationNotPermitted, err)
+	})
+	t.Run("address is not in same shard as the caller", func(t *testing.T) {
+		t.Parallel()
+
+		input := &vmcommon.ContractCallInput{
+			VMInput: vmcommon.VMInput{
+				CallValue:  big.NewInt(0),
+				Arguments:  [][]byte{[]byte("arg1")},
+				CallerAddr: []byte("caller"),
+			},
+			RecipientAddr: []byte("caller"),
+		}
+		shardCoordinator := &mock.ShardCoordinatorStub{
+			SameShardCalled: func(firstAddress, secondAddress []byte) bool {
+				return false
```

Comment:
add 2 asserts here for the firstAddress and secondAddress that should be the same with CallerAddr and Arguments[0] ?

Reply:
Added.